HEN_HOUSE/egs++/geometry/egs_autoenvelope/egs_autoenvelope.cpp

```diff
@@ -0,0 +1,1018 @@
+    EGS_BaseGeometry(Name), base_geom(base_geom), debug_info(debug), output_vc(output_vc_file) {
+
+    base_geom->ref();
+    nregbase = base_geom->regions();
+    is_convex = base_geom->isConvex();
+    has_rho_scaling = getHasRhoScaling();
+
+    // initialize uncorrected/corrected masses
+    for (int ir = 0; ir < nregbase; ir++) {
+        uncorrected_mass.push_back(base_geom->getMass(ir));
+        corrected_mass.push_back(base_geom->getMass(ir));
+    }
+
+    ninscribed = inscribed.size();
+    if (ninscribed == 0) {
+        egsFatal("EGS_AEnvelope: no inscribed geometries!\n");
+    }
+
+    // nreg is total regions in geometry = nregbase + sum(nreginscribed_j)
+    nreg = nregbase;
+
+    // create a transformed copy of the inscribed geometry
+    int global = nreg;
+    for (int geom_idx=0; geom_idx < ninscribed; geom_idx++) {
+
+        EGS_BaseGeometry *inscribed_geom = inscribed[geom_idx].geom;
+        EGS_AffineTransform *transform = inscribed[geom_idx].transform;
+        VCOptions *vcopt = inscribed[geom_idx].vcopts;
+        EGS_TransformedGeometry *transformed = new EGS_TransformedGeometry(inscribed_geom, *transform);
+
+        inscribed_geoms.push_back(transformed);
+        transforms.push_back(transform);
+        opts.push_back(vcopt);
+
+        int ninscribed_reg= transformed->regions();
+        nreg += ninscribed_reg;
+
+        for (int lreg = 0; lreg < ninscribed_reg; lreg++) {
+            GeomRegPairT local(transformed, lreg);
+            local_to_global_reg[local] = global;
+            global_reg_to_local[global] = local;
+            global++;
+        }
+    }
+
+    geoms_in_region = new vector<EGS_BaseGeometry *>[nregbase];
+
+    if (inscribed[0].vcopts->vc_file != "") {
+        vc_results = loadFileResults(inscribed[0].vcopts, base_geom, inscribed_geoms, transforms);
+        egsInformation("loaded from %s\n", inscribed[0].vcopts->vc_file.c_str());
+    }
+    else {
+        // now run volume correction and figure out which regions have inscribed geometries
+        vc_results = findRegionsWithInscribed(inscribed[0].vcopts, base_geom, inscribed_geoms, transforms);
+    }
+
+
+    nreg_with_inscribed = 0;
+    for (volcor::RegionGeomSetT::iterator it=vc_results.regions_with_inscribed.begin();
+            it!=vc_results.regions_with_inscribed.end(); it++) {
+        nreg_with_inscribed++;
+        copy(it->second.begin(), it->second.end(), back_inserter(geoms_in_region[it->first]));
+    }
+
+    // set mass correction based on volume correction
+    for (int ir = 0; ir < nregbase; ir++) {
+        double volume_ratio = vc_results.corrected_volumes[ir]/vc_results.uncorrected_volumes[ir];
+        corrected_mass[ir] = uncorrected_mass[ir]*volume_ratio;
+    }
+
+    if (getNRegWithInscribed() == 0) {
+        egsFatal("EGS_AEnvelope:: Failed to find any regions with inscribed geometries\n");
+    }
+
+    if (output_vc=="text" || output_vc=="gzip") {
```

Comment:
We're also supposed to be able to use "yes" as equivalent to "text", so this line should be:

`if (output_vc=="yes" || output_vc=="text" || output_vc=="gzip") {`